### PR TITLE
ASP-2106 Add download_application_files to the CLI

### PR DESCRIPTION
**Pull request by fschuch**
#### What

* A `download-job-script` command is added to the CLI
  * Takes at a minimum the job_script_id to download
  * All job_scripts associated with the entry are downloaded to the CWD
  * Output is supplied to the user providing the paths of the downloaded files
* A `download-application` command is added to the CLI
  * Takes at a minimum the application_id to download
  * All files associated with the entry are downloaded to the CWD
  * Output is supplied to the user providing the paths of the downloaded files
* Help entries are included describing the commands and their arguments
* Unit tests are included for both commands.

#### Why
As a Jobbergate CLI user, I need to be able to download the files associated with a job_script or application entry so that I can review the source code outside of the CLI

`Task`: https://jira.scania.com/browse/ASP-2106

---

#### Peer Review
Please follow the upstream omnivector documentation concerning
[peer-review guidelines](https://github.com/omnivector-solutions/Documentation/blob/main/Contributing/pr_review_standards.md#peer-review).

### CODE REVIEW

**Comment by fschuch** [comment]
The fix on the CLI's tests was done on https://github.com/omnivector-solutions/jobbergate/pull/194, so it will be fixed here once the cited PR is merged and this branch is rebased with `main`.

File: jobbergate-cli/jobbergate_cli/subapps/applications/tools.py [hunk 1]
```diff
@@ -185,6 +190,39 @@ def upload_application(
         return response_code == 201
 
 
+def download_application_files(
```
Comment (fschuch) [inline jobbergate-cli/jobbergate_cli/subapps/applications/tools.py:193]: This function would be better named `save_application_files`

File: jobbergate-cli/jobbergate_cli/subapps/job_scripts/tools.py [hunk 2]
```diff
@@ -127,3 +129,23 @@ def create_job_script(
     )
 
     return job_script_result
+
+
+def download_job_script_files(
```
Comment (fschuch) [inline jobbergate-cli/jobbergate_cli/subapps/job_scripts/tools.py:134]: This function would be better named `save_job_script_files`

File: jobbergate-cli/tests/subapps/job_scripts/test_app.py [hunk 3]
```diff
@@ -16,8 +19,11 @@
     JOB_SUBMISSION_HIDDEN_FIELDS,
     create,
     delete,
+    download_files,
     get_one,
+    handle_abort,
```
Comment (fschuch) [inline jobbergate-cli/tests/subapps/job_scripts/test_app.py:24]: This import is unised

File: jobbergate-cli/jobbergate_cli/subapps/applications/tools.py (hunk 1 above)
Comment (fschuch) [reply jobbergate-cli/jobbergate_cli/subapps/applications/tools.py]: solved in [802a6e0](https://github.com/omnivector-solutions/jobbergate/pull/198/commits/802a6e0dbdd5c472b04832387af5b93e73949eb9)

File: jobbergate-cli/jobbergate_cli/subapps/job_scripts/tools.py (hunk 2 above)
Comment (fschuch) [reply jobbergate-cli/jobbergate_cli/subapps/job_scripts/tools.py]: solved in [802a6e0](https://github.com/omnivector-solutions/jobbergate/pull/198/commits/802a6e0dbdd5c472b04832387af5b93e73949eb9)

File: jobbergate-cli/tests/subapps/job_scripts/test_app.py [hunk 4]
```diff
@@ -1,9 +1,12 @@
 import importlib
 import json
 import shlex
+from unittest import mock
 
 import httpx
+import pytest
 
+from jobbergate_cli.exceptions import Abort
```
[the remark this answers is not in the document]
Comment (fschuch) [reply jobbergate-cli/tests/subapps/job_scripts/test_app.py]: solved in [171c3db](https://github.com/omnivector-solutions/jobbergate/pull/198/commits/171c3db88b03ec2dad7a85c542f5fe99692c8eb3)

File: jobbergate-cli/tests/subapps/job_scripts/test_app.py (hunk 3 above)
Comment (fschuch) [reply jobbergate-cli/tests/subapps/job_scripts/test_app.py]: solved in [171c3db](https://github.com/omnivector-solutions/jobbergate/pull/198/commits/171c3db88b03ec2dad7a85c542f5fe99692c8eb3)

File: jobbergate-cli/jobbergate_cli/subapps/applications/app.py [hunk 5]
```diff
@@ -314,3 +320,37 @@ def delete(
...
+    ),
+):
+    """
+    Download the files from an application to the current working directory.
+    """
+    jg_ctx: JobbergateContext = ctx.obj
+    result = fetch_application_data(jg_ctx, id=id, identifier=identifier)
+    saved_files = save_application_files(
+        application_data=result,
+        destination_path=pathlib.Path.cwd(),
```
Comment (matheushent) [inline jobbergate-cli/jobbergate_cli/subapps/applications/app.py:345]: shouldn't this path be an input from the user?

Comment (fschuch) [reply jobbergate-cli/jobbergate_cli/subapps/applications/app.py:345]: That could be done, but the current working directory was specified in the requirements.

**Review by dusktreader — APPROVED** [review]

**Pull request merged**